app/scripts/services/discovery.js

```diff
@@ -1,11 +1,16 @@
 'use strict';
 
 angular.module('openshiftConsole')
-  .factory('APIDiscovery', ['LOGGING_URL', 'METRICS_URL', '$q', function(LOGGING_URL, METRICS_URL, $q) {
+  .factory('APIDiscovery', ['LOGGING_URL', 'METRICS_URL', '$q', '$filter', function(LOGGING_URL, METRICS_URL, $q, $filter) {
     return {
       // Simulate asynchronous requests for now. If these are ever updated to call to a discovery
       // endpoint, we need to make sure to trigger a digest loop using (or update all callers).
-      getLoggingURL: function() {
+      getLoggingURL: function(project) {
+        loggingUIHostname = $filter('annotationName')(project, 'loggingUIHostname');
+        if(loggingUIHostname) {
+           return 'https://' + loggingUIHostname;
```

Comment:
Sorry @jcantrill missed this on last review, but this needs to be wrapped in a `$q.when` since the method returns a promise.

```js
return $q.when('https://' + loggingUIHostname);
```

Or you can change it to have one return.

```js
var loggingURL = LOGGING_URL;
var loggingUIHostname = $filter('annotationName')(project, 'loggingUIHostname');
if(loggingUIHostname) {
  loggingURL = 'https://' + loggingUIHostname;
}

return $q.when(loggingURL);
```

Reply:
fixed